ivy/functional/ivy/experimental/linear_algebra.py

```diff
@@ -1666,3 +1666,86 @@ def dot(
     ivy.array([[-15.28]])
     """
     return current_backend(a, b).dot(a, b, out=out)
+
+
+# This function has been adapted from TensorLy
+# https://github.com/tensorly/tensorly/blob/main/tensorly/tenalg/core_tenalg/moments.py#L5
+
+
+@handle_nestable
+@handle_exceptions
+@handle_array_like_without_promotion
+@inputs_to_ivy_arrays
+@handle_array_function
+@handle_device_shifting
+def higher_order_moment(
+    tensor: Union[ivy.Array, ivy.NativeArray], order: Optional[int], /
+) -> ivy.Array:
+    """
+    Compute the Higher-Order Moment.
+
+    Parameters
+    ----------
+    tensor : 2D-tensor -- or ND-tensor
+        matrix of size (n_samples, n_features)
+        or tensor of size(n_samples, D1, ..., DN)
+
+    order : int
+        order of the higher-order moment to compute
+
+    Returns
+    -------
+    tensor : moment
+        if tensor is a matrix of size (n_samples, n_features),
+        tensor of size (n_features, )*order
+    """
+    moment = ivy.copy_array(tensor)
+    for _ in range(order - 1):
+        moment = _batched_outer([moment, tensor])
+
+    return ivy.mean(moment, axis=0)
+
+
+def _batched_outer(tensors):
+    """
+    Return a generalized outer product of the tensors.
+
+    Parameters
+    ----------
+    tensors : list of tensors
+        list of tensors of shape (n_samples, J1, ..., JN) ,
+        (n_samples, K1, ..., KM) ...
+
+    Returns
+    -------
+    outer product of tensors
+        of shape (n_samples, J1, ..., JN, K1, ..., KM, ...)
+    """
+    result = None
+    result_size = None
+    result_shape = None
+    for i, tensor in enumerate(tensors):
+        if i:
+            current_shape = ivy.shape(tensor)
+            current_size = len(current_shape) - 1
+
+            n_samples = current_shape[0]
+
+            if n_samples != result_shape[0]:
+                raise ValueError(
+                    f"Tensor {i} has a batch-size of {n_samples} but those before had a"
+                    f" batch-size of {result_shape[0]}, all tensors should have the"
+                    " same batch-size."
```

Comment:
I think I missed this function while creating the ToDo list but I believe this is a general function and should be added to the functional API of Ivy. I have added it to the ToDo list now (https://github.com/unifyai/ivy/issues/21914). Do you mind working on this function as well? You can add both functions to this PR itself.